paimon-trino-common/src/main/java/org/apache/paimon/trino/TrinoMetadata.java

```diff
@@ -448,7 +461,37 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(
                                 .intersect(TupleDomain.withColumnDomains(unenforcedDomains));
 
         return Optional.of(
-                new ConstraintApplicationResult<>(trinoTableHandle.copy(newFilter), remain, false));
+                new ConstraintApplicationResult<>(
+                        trinoTableHandle.copy(newFilter, expressionFilter), remain, false));
+    }
+
+    private static TupleDomain<TrinoColumnHandle> getTrinoColumnHandleTupleDomain(
+            Constraint constraint) {
+        Map<TrinoColumnHandle, Domain> expressionPredicates = new HashMap<>();
```

Comment:
I think we could just declair expressionPredicate. In default: we put expressionPredicates=Collections.emptyMap()